test/e2e/testsuites/testsuites.go

```diff
@@ -1208,6 +1199,31 @@ func (t *TestAzVolumeAttachment) WaitForReplicas(numReplica int, timeout time.Du
 	return wait.PollImmediate(time.Duration(15)*time.Second, timeout, conditionFunc)
 }
 
+func CreateNodeFailure(client v1alpha1ClientSet.DiskV1alpha1Interface, namespace, nodeName string) error {
+	err := client.AzDriverNodes(namespace).Delete(context.Background(), nodeName, metav1.DeleteOptions{})
+	framework.Logf("Created Failure on AzDriverNode for Node (%s)", nodeName)
+	return err
+}
+
+func RecoverNodeFailure(client v1alpha1ClientSet.DiskV1alpha1Interface, namespace, nodeName string) {
```

Comment:
Same goes for this method. The functionality is the same as `NewTestAzDriverNode`. You could add a comment for instances when these methods are used to create or recover from failure. I would prefer to avoid replication because CRD definitions change and it would be good to limit the places where we need to make manual updates for those changes.